panic in `juju status --storage`

### Description

```
$ juju status --storage
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x221de4c]

goroutine 1 [running]:
github.com/juju/juju/cmd/juju/common.FormatTime(0x13a3c8c789f0?, 0x10?)
	/home/jack/juju/juju/cmd/juju/common/format.go:103 +0x8c
github.com/juju/juju/cmd/juju/storage.createStorageInfo({{0x13a3c8c789f0, 0x10}, {0x0, 0x0}, 0x2, {{0x13a3c8c78a00, 0x7}, {0x0, 0x0}, 0x0, ...}, ...})
	/home/jack/juju/juju/cmd/juju/storage/storage.go:97 +0xcd
github.com/juju/juju/cmd/juju/storage.formatStorageDetails({0x13a3c8541860, 0x2, 0x13a3c8bdc840?})
	/home/jack/juju/juju/cmd/juju/storage/storage.go:75 +0x1b3
github.com/juju/juju/cmd/juju/storage.CombinedStorageFromParams({0x13a3c8541860, 0x2, 0x2}, {0x13a3c8e3aa00, 0x1, 0x1}, {0x13a3c8e40000, 0x2, 0x2})
	/home/jack/juju/juju/cmd/juju/storage/list.go:179 +0x65
github.com/juju/juju/cmd/juju/status.(*statusCommand).runStatus(0x13a3c8bdc840, 0x13a3c85b2d20)
	/home/jack/juju/juju/cmd/juju/status/status.go:282 +0x3af
github.com/juju/juju/cmd/juju/status.(*statusCommand).Run(0x13a3c8f0c690?, 0x6abb518?)
	/home/jack/juju/juju/cmd/juju/status/status.go:342 +0x49
github.com/juju/juju/cmd/modelcmd.(*modelCommandWrapper).Run(0x13a3c8f0c690, 0x13a3c85b2d20)
	/home/jack/juju/juju/cmd/modelcmd/modelcommand.go:649 +0x124
github.com/juju/juju/cmd/modelcmd.(*baseCommandWrapper).Run(0x13a3c89af420, 0x13a3c85b2d20)
	/home/jack/juju/juju/cmd/modelcmd/base.go:572 +0x95
github.com/juju/juju/cmd/cmd.(*SuperCommand).Run(0x13a3c8bddb80, 0x13a3c85b2d20)
	/home/jack/juju/juju/cmd/cmd/supercommand.go:562 +0x389
github.com/juju/juju/cmd/cmd.Main({0x6ac69d0, 0x13a3c8bddb80}, 0x13a3c85b2d20, {0x13a3c8d74000, 0x2, 0x2})
	/home/jack/juju/juju/cmd/cmd/cmd.go:450 +0x232
github.com/juju/juju/cmd/juju/commands.jujuMain.Run({0x13a3c869c3c0?}, {0x13a3c8392210, 0x3, 0x3})
	/home/jack/juju/juju/cmd/juju/commands/main.go:195 +0x86d
github.com/juju/juju/cmd/juju/commands.Main(...)
	/home/jack/juju/juju/cmd/juju/commands/main.go:119
main.main()
```

### Juju version

4.0.6

### Cloud

AWS

### Expected behaviour

No panic

### Reproduce / Test

```
$ juju deploy postgresql --storage pgdata=50G sql -n 2
$ juju remove-storage pgdata/0 --force
$ juju status --storage
```

### Notes & References

_No response_

I confirm the issue is still here despite it was claimed to be fixed in https://github.com/juju/juju/issues/22024

```
ubuntu@juju40:~$ juju status --color --storage --relations
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2231b4c]

goroutine 1 [running]:
github.com/juju/juju/cmd/juju/common.FormatTime(0xc0001e7d20?, 0xe?)
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/common/format.go:103 +0x8c
github.com/juju/juju/cmd/juju/storage.createStorageInfo({{0xc0001e7d20, 0xe}, {0x0, 0x0}, 0x2, {{0xc0001e7d15, 0x7}, {0x0, 0x0}, 0x0, ...}, ...})
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/storage/storage.go:97 +0xbc
github.com/juju/juju/cmd/juju/storage.formatStorageDetails({0xc0007e8780, 0x4, 0xc00063a9a0?})
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/storage/storage.go:75 +0x158
github.com/juju/juju/cmd/juju/storage.CombinedStorageFromParams({0xc0007e8780, 0x4, 0x4}, {0xc000259400, 0x2, 0x2}, {0x0, 0x0, 0x0})
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/storage/list.go:179 +0x65
github.com/juju/juju/cmd/juju/status.(*statusCommand).runStatus(0xc00063a9a0, 0xc0002a09c0)
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/status/status.go:316 +0x3b8
github.com/juju/juju/cmd/juju/status.(*statusCommand).Run(0xc0008daf30?, 0x704c428?)
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/status/status.go:376 +0x49
github.com/juju/juju/cmd/modelcmd.(*modelCommandWrapper).Run(0xc0008daf30, 0xc0002a09c0)
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/modelcmd/modelcommand.go:649 +0x124
github.com/juju/juju/cmd/modelcmd.(*baseCommandWrapper).Run(0xc000b00fd0, 0xc0002a09c0)
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/modelcmd/base.go:572 +0x95
github.com/juju/juju/cmd/cmd.(*SuperCommand).Run(0xc00063bce0, 0xc0002a09c0)
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/cmd/supercommand.go:562 +0x38b
github.com/juju/juju/cmd/cmd.Main({0x7053d30, 0xc00063bce0}, 0xc0002a09c0, {0xc0002f19c0, 0x4, 0x4})
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/cmd/cmd.go:449 +0x232
github.com/juju/juju/cmd/juju/commands.jujuMain.Run({0xc0002f03c0?}, {0xc0000b4050, 0x5, 0x5})
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/commands/main.go:195 +0x8a7
github.com/juju/juju/cmd/juju/commands.Main(...)
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/commands/main.go:119
main.main()
        /build/snapcraft-juju-31c6fc4d927733460798d76df40c02ca/parts/juju/build/cmd/juju/main.go:29 +0x6c


ubuntu@juju40:~$ juju version
4.0.5-genericlinux-amd64  # LXD 
```

--- a/apiserver/facades/client/storage/storage.go
+++ b/apiserver/facades/client/storage/storage.go
@@ -1310,8 +1310,8 @@ func (a *StorageAPI) removeStorageInstance(
 			err := a.detachStorageAttachment(ctx, saUUID, force, wait)
 			if err != nil {
 				return errors.Errorf(
-					"removing storage attachment %q for storage %q:",
-					saUUID, tag.Id(),
+					"removing storage attachment %q for storage %q: %w",
+					saUUID, tag.Id(), err,
 				)
 			}
 		}

--- a/apiserver/facades/client/storage/storageremove_test.go
+++ b/apiserver/facades/client/storage/storageremove_test.go
@@ -10,6 +10,8 @@ import (
 	"github.com/juju/tc"
 	"go.uber.org/mock/gomock"
 
+	coreunit "github.com/juju/juju/core/unit"
+	applicationerrors "github.com/juju/juju/domain/application/errors"
 	domainstorage "github.com/juju/juju/domain/storage"
 	storageerrors "github.com/juju/juju/domain/storage/errors"
 	"github.com/juju/juju/rpc/params"
@@ -190,4 +192,53 @@ func (s *storageRemoveSuite) TestRemoveWithStorageAttachments(c *tc.C) {
 	c.Assert(err, tc.ErrorIsNil)
 	c.Assert(res.Results, tc.HasLen, 1)
 	c.Check(res.Results[0].Error, tc.IsNil)
+}
+
+// TestRemoveWithStorageAttachmentErrorIncludesCause checks that
+// removeStorageInstance preserves the reason for an attachment removal failure.
+// The regression returned only the outer context with a trailing colon.
+func (s *storageRemoveSuite) TestRemoveWithStorageAttachmentErrorIncludesCause(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	storageInstanceUUID := tc.Must(c, domainstorage.NewStorageInstanceUUID)
+	attachmentUUID := tc.Must(c, domainstorage.NewStorageAttachmentUUID)
+	unitUUID := tc.Must(c, coreunit.NewUUID)
+
+	storageExp := s.storageService.EXPECT()
+	storageExp.GetStorageInstanceUUIDForID(
+		gomock.Any(), "data/1",
+	).Return(storageInstanceUUID, nil)
+	storageExp.GetStorageInstanceAttachments(
+		gomock.Any(),
+		storageInstanceUUID,
+	).Return([]domainstorage.StorageAttachmentUUID{
+		attachmentUUID,
+	}, nil)
+
+	removalExp := s.removalService.EXPECT()
+	removalExp.RemoveStorageAttachment(
+		gomock.Any(), attachmentUUID, true, time.Duration(0),
+	).Return("", applicationerrors.UnitStorageMinViolation{
+		CharmStorageName: "data",
+		RequiredMinimum:  1,
+		UnitUUID:         unitUUID.String(),
+	})
+
+	api := s.makeTestAPIForIAASModel(c)
+	res, err := api.Remove(c.Context(), params.RemoveStorage{
+		Storage: []params.RemoveStorageInstance{
+			{
+				Tag:                "storage-data/1",
+				DestroyAttachments: true,
+				Force:              new(true),
+			},
+		},
+	})
+	c.Assert(err, tc.ErrorIsNil)
+	c.Assert(res.Results, tc.HasLen, 1)
+	c.Check(res.Results[0].Error, tc.DeepEquals, &params.Error{
+		Code: params.CodeNotValid,
+		Message: `removing storage attachment "` + attachmentUUID.String() +
+			`" for storage "data/1": removing storage from unit would violate charm storage "data" requirements of having minimum 1 storage instances`,
+	})
 }

